frontend/src/components/widgets/UserAvatarLink.jsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import Avatar from "@material-ui/core/Avatar";
+import Typography from "@material-ui/core/Typography";
+import { makeStyles } from "@material-ui/styles";
+import { getUrl as getUserBoardUrl } from "./LinkToUserBoard";
+
+const useStyles = makeStyles({
+  avatarLink: {
+    textDecoration: "none",
+  },
+  container: {
+    display: "flex",
+    alignItems: "center",
+    gap: 8,
+  },
+});
+
+function getAvatarTextInitials(email) {
+  // For Umuzi emails - take the first letters of the name and surname
+  if (email.endsWith("umuzi.org")) {
+    const [firstname, lastname] = email.split(".");
```

Comment:
I don't know if this is an issue, but if we were to ever want to use the last name for something else, this variable `lastname` would have to be broken down further. for example, for `ngoako.ramokgopa@umuzi.org` the lastname is ramokgopa@umuzi. but even if not, I think the variable name has to tell the truth about its value.